frontend/public/components/utils/safe-fetch-hook.ts

```diff
@@ -0,0 +1,13 @@
+
+import { useEffect, useRef } from 'react';
+import { coFetchJSON } from '../../co-fetch';
+
+export const useSafeFetch = () => {
+  const controller = useRef<AbortController>();
+  useEffect(() => {
+    controller.current = new AbortController();
+    return () => controller.current.abort();
+  }, []);
+
+  return (url) => coFetchJSON(url, 'get', {signal: controller.current.signal});
```

Comment:
This is really awesome. Would be nice to have a unit test for it, you can check out some of the other unit tests for hooks that we've written. Also, we may want to use this for other HTTP methods in the future, but not needed now.

Reply:
Adding a unit test for this would be nice, but I went ahead and added `lgtm` since this PR is blocking some other graph work.